sdks/python/apache_beam/testing/load_tests/sideinput_test.py

```diff
@@ -0,0 +1,158 @@
+
+"""
+
+from __future__ import absolute_import
+
+import json
+import logging
+import unittest
+
+import apache_beam as beam
+from apache_beam.pvalue import AsIter
+from apache_beam.testing import synthetic_pipeline
+from apache_beam.testing.load_tests.load_test_metrics_utils import MeasureTime
+from apache_beam.testing.test_pipeline import TestPipeline
+
+
+class SideInputTest(unittest.TestCase):
+  def _parseTestPipelineOptions(self):
+    return {
+        'numRecords': self.inputOptions.get('num_records'),
+        'keySizeBytes': self.inputOptions.get('key_size'),
+        'valueSizeBytes': self.inputOptions.get('value_size'),
+        'bundleSizeDistribution': {
+            'type': self.inputOptions.get(
+                'bundle_size_distribution_type', 'const'
+            ),
+            'param': self.inputOptions.get('bundle_size_distribution_param', 0)
+        },
+        'forceNumInitialBundles': self.inputOptions.get(
+            'force_initial_num_bundles', 0
+        )
+    }
+
+  def _getSideInput(self):
+    side_input = self._parseTestPipelineOptions()
+    side_input['numRecords'] = side_input['numRecords']
+    side_input['keySizeBytes'] = side_input['keySizeBytes']
+    side_input['valueSizeBytes'] = side_input['valueSizeBytes']
+    return side_input
+
+  def _getPerElementDelaySec(self):
+    return self.syntheticStepOptions.get('per_element_delay_sec', 0)
+
+  def _getPerBundleDelaySec(self):
+    return self.syntheticStepOptions.get('per_bundle_delay_sec', 0)
+
+  def _getOutputRecordsPerInputRecords(self):
+    return self.syntheticStepOptions.get('output_records_per_input_records', 0)
+
+  def setUp(self):
+    self.pipeline = TestPipeline()
+    self.inputOptions = json.loads(self.pipeline.get_option('input_options'))
+    self.iterations = self.pipeline.get_option('number_of_counter_operations')
+    if self.iterations is None:
+      self.iterations = 1
+
+  class Consume(beam.DoFn):
+    def process(self, element, *args, **kwargs):
+      values_list = []
+      for i in element:
+        for _, value in i.iteritems():
+          values_list.append(value)
+      yield values_list
+
+  def testSideInput(self):
+    def join_fn(element, side_input, iterations):
+      list = []
+      for i in range(iterations):
+        for key, value in side_input:
+          if i == iterations - 1:
+            list.append({key: element[1]+value})
+      yield list
+
+    with self.pipeline as p:
+      main_input = (p
+                    | "Read pcoll 1" >> beam.io.Read(
+                        synthetic_pipeline.SyntheticSource(
+                            self._parseTestPipelineOptions()))
+                   )
+
+      side_input = (p
+                    | "Read pcoll 2" >> beam.io.Read(
+                        synthetic_pipeline.SyntheticSource(
+                            self._getSideInput()))
+                   )
+      # pylint: disable=expression-not-assigned
+      (main_input
+       | "Merge" >> beam.ParDo(
```

Comment:
By iterating through the side input in `"Merge"`, you will consume it, and no longer need to iterate through it in `"Consume"` - so it may be fine if you just drop the `Consume` transform. Other than that, I think this must be fine : )

Reply:
That sounds good. Thanks! `Consume` was useful to check if `join_fn` correctly merges together `main_input` and `side input`. But you are right - now it's not necessary.